src/components/Variants/VariantList.jsx

```diff
@@ -0,0 +1,34 @@
+import { useEffect, useState } from "react";
+import VariantItem from "./VariantItem.jsx";
+
+const VariantList = () => {
+  const [variantOptions, setVariantOptions] = useState([]);
+
+  useEffect(() => {
+    const setup = async () => {
+      const viewerDom = document.querySelector("needle-engine");
+      const context = await viewerDom?.getContext();
+      setTimeout(() => {
```

Comment:
Curious, timeout needed for async oddities?

Reply:
Yeah, I asked Xiao what to wait for to know when the variantOptions are done loading since that's after the context loads, and there isn't a good way right now.